File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/java/field_assign_to_array_item_in_function.java
class Foo {
    String field;
}
class A {
    void func(Foo fo111, String[] fo222)
    {
        fo111.field = fo222[1];
    }

    int main()
    {
        Foo f11;
        String[] f22 = new String[10];
        f22[1] = null;
        func(f11, f22);
        return f11.field.length();  // should be detected here
    }
}

File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/field_assign_to_array_item_in_function.cpp
class Foo {
public:
    int *field;
    int *other;
};

void func(Foo* fo111, int* fo222[10]) {
    fo111->field = fo222[1];
}

int main() {
    Foo f11;
    int* f22[10];
    f22[1] = 0;
    func(&f11, f22);
    return *f11.field; // should be detected here
}
